src/test/java/com/giovds/DependenciesToQueryMapperTest.java

```diff
@@ -25,15 +31,18 @@ void should_return_lucene_query_based_on_given_dependencies() {
     @Test
     void should_return_multiple_lucene_query_based_on_given_dependencies_when_exceeding_limit() {
```

Comment:
The additions here are awesome 🔥

I was playing around with the `IntStream` and stumbled on a `StringIndexOutOfBoundsException` when the amount of dependencies is dividable by the `maxItemsPerQuery`. The final iteration results in an empty query String when trying to remove the `+OR+` at the end of the query. I actually think this happens in the currently implementation as well, but it just very unlikely to happen due to the more random nature of the query length 🙈

```java
@ParameterizedTest
@ValueSource(ints = {25, 30, 40, 59, 60})
void should_return_multiple_lucene_query_based_on_given_dependencies_when_exceeding_limit(final int numberOfDeps) {
    final var division = (int) Math.ceil((double) numberOfDeps / DependenciesToQueryMapper.maxItemsPerQuery);
    final var otherDeps = IntStream.rangeClosed(1, numberOfDeps)
            .mapToObj(i -> String.format("bar-artifact-%d", i))
            .map(artifactId -> createDependency("org.apache.maven", artifactId, "1.2.3"))
            .collect(Collectors.toSet());

    final List<String> actualQueries = DependenciesToQueryMapper.mapToQueries(otherDeps);

    assertThat(actualQueries).hasSize(division).allMatch(query -> query.endsWith(")"));
}
```

Reply:
Nice catch. I've updated the test, made the query batching more robust, and added a few more edge cases for the number of dependencies.